dashboard/src/modules/components/RelayUIComponent/index.jsx

```diff
@@ -73,9 +73,14 @@ const RelayComponent = () => {
       onClose={handleClose}
     >
       <div className="modal-heading">
-        <Title headingLevel="h3">Relay</Title>
+        <Title headingLevel="h3">Pull Datasets</Title>
         <PopoverComponent />
       </div>
+      <p className="modal-desc">
+        The Pbench Agent can push datasets to a Pbench Server directly, when the
+        server is accessible from the system where the Pbench Agent is being
+        used.
```

Comment:
The "relay" is for the opposite case, when the server *isn't* accessible from the Pbench Agent host. Maybe something like

> When the Pbench Server is not reachable from the system where the Pbench Agent is being used, Pbench Agent can instead upload results data to a relay server that is reachable from both the Pbench Agent and the Pbench Server. The Pbench Agent `pbench-results-move` command when used with the `--relay` option will report a URL on successful upload, and entering that URL here will instruct the Pbench Server to pull the results data from the relay and then remove it from the relay.

Reply:
I second Dave, here.  While I like his text, I would tweak it a bit:
>When the Pbench Server is not directly reachable from the system where the Pbench Agent is being used, the Pbench Agent can instead upload results data to a relay server that is reachable by both the Pbench Agent and the Pbench Server.  When used with the `--relay` option, the Pbench Agent `pbench-results-move` command will report a URL on successful upload, and entering that URL here will instruct the Pbench Server to pull the results data from the relay and then remove it from the relay.

Reply:
Although it occurs to me that we could (if we want to be wordy) go beyond even "reachable" ... you can push directly to a server *iff* you have an API Key for the server and you're capable of transmitting data through an `https` connection. But the relay might also be useful in some cases just because it allows you to delay the Pbench Server authentication until you're at the dashboard where you can just log in and don't need to worry about carrying an API key in your pocket?

(We don't, however, want to get *too* fancy in this little hint, so maybe not!)

Reply:
> The word "directly" doesn't help, and arguably is incorrect: either it's "reachable" or not.

While that is certainly true in a technical sense, there's often a trade-off between precision and clarity, and I think that introducing the concept of a "direct" connection helps provide context even if it is technically superfluous.  (To my way of thinking, if the Server weren't at least indirectly reachable from the Agent, then there would be no way that a relay could work; if it's directly reachable, then there is no need for a relay (modulo what you pointed out); thus, the relay is required when the Agent can reach an intermediary which the Server can also reach, which sounds like a connection but not a direct one, to me.)

But, it sounds like you feel strongly, so I'm fine with foregoing "directly".

> I don't particularly like your phrase switch on the `--relay` option since I'd rather put the actor (the command) front and center; but that's a nit.

I felt like the `--relay` phrase was interrupting the flow, so I moved it to the front, but I don't care strongly.

> Although it occurs to me that we could (if we want to be wordy) go beyond even "reachable" ... you can push directly to a server _iff_ you have an API Key for the server and you're capable of transmitting data through an `https` connection. But the relay might also be useful in some cases just because it allows you to delay the Pbench Server authentication until you're at the dashboard where you can just log in and don't need to worry about carrying an API key in your pocket?

I am attracted to the use case that this enables where the user doesn't have to generate, distribute, store, and protect a token (if they prefer not to), since they can pull the result using their login credentials instead.  However, I'm not sure that the "pull description" here is the right place to describe this.  (Perhaps, the documentation for `pbench-results-push`, particularly the description of the `--token` option and/or the `--relay` option would be a better location.)

What we need this text to explain is that the user has the option to "pull" the result, instead of pushing it, and where to get the "pull" URL from...and the answer to that is `pbench-results-push`.  It's possible that we don't have to explain very much more than that here.  (We don't necessarily have to explain the "why"; we just need to point out the "what" and "how".)  The documentation for `pbench-results-push` can describe the choices of pushing to the Server vs. pushing to a relay along with their respective requirements and why you might opt for one over the other.

So, perhaps something more like this would be better:
> The Pbench Agent `pbench-results-move` command when used with the `--relay` option will report a URL on successful upload.  Enter that URL here to instruct the Pbench Server to upload the results data from the relay and remove it from the relay.

or

> Instead of using the Pbench Agent `pbench-results-move` command to push results directly to the Pbench Server, the Server can pull properly prepared results from a relay.  Specify the `--relay` option to the `pbench-results-move` command and then enter here the URL which it prints out.  This will cause the Server to upload those results and remove them from the relay.

Reply:
> What we need this text to explain is that the user has the option to "pull" the result, instead of pushing it, and where to get the "pull" URL from...and the answer to that is `pbench-results-push`. It's possible that we don't have to explain very much more than that here. (We don't necessarily have to explain the "why"; we just need to point out the "what" and "how".)

💯 

So a bit simplified, riffing only slightly on your second-last submission:

> When you use the Pbench Agent `pbench-results-move` command the `--relay` option to push results to a relay server, the command will report a URL. Enter that URL here to instruct the Pbench Server to upload the results data from the relay and remove it from the relay.